src/xtgeo/common/sys.py

```diff
@@ -561,48 +547,53 @@ def detect_fformat(
         #   3        2016           0           0           0           0
         #  (ver)    (release)      (reserved)   (backw)    (gtype)      (dualporo)
 
-        fort1, name, num, _, fort2 = struct.unpack("> i 8s i 4s i", buf[:24])
-        if fort1 == 16 and name == b"FILEHEAD" and num == 100 and fort2 == 16:
-            # Eclipse corner point EGRID
-            logger.info("Signature is egrid")
-            return self._validate_format("egrid")
+        if len(buf) >= 24:
+            fort1, name, num, _, fort2 = struct.unpack("> i 8s i 4s i", buf[:24])
+            if fort1 == 16 and name == b"FILEHEAD" and num == 100 and fort2 == 16:
```

Comment:
There is a lot of repetition in the 3 following sections, it could be collapsed.

Reply:
collapsed as suggested.